src/resolve/resolved-dns-transaction.c

```diff
@@ -1816,7 +1816,8 @@ static int dns_transaction_prepare(DnsTransaction *t, usec_t ts) {
                 if (r > 0) {
                         dns_transaction_randomize_answer(t);
 
-                        if (t->bypass && t->scope->protocol == DNS_PROTOCOL_DNS && !t->received)
+                        if (t->bypass && t->scope->protocol == DNS_PROTOCOL_DNS && (!t->received ||
+                                                !DNS_PACKET_CD(t->bypass)))
```

Comment:
Spurious line wrapping...
The line is not so long, please drop the line break.
Or, at least, please wrap the line after `&&`.